protocols/http/http-client.js

```diff
@@ -1,12 +1,20 @@
 var peer = Math.random().toString(36).substr(2)
 var enable_cors_default = false
 
+if (typeof module !== 'undefined' && module.exports) {
+    module.exports = braid_fetch
+}
+
+var fetch
+var Headers
 // On nodejs, this requires "npm install node-fetch node-web-streams"
 if (typeof window === 'undefined') {
-    var fetch = require('node-fetch')
-    var Headers = fetch.Headers
+    fetch = require('node-fetch')
+    Headers = fetch.Headers
     var to_whatwg_stream = require('node-web-streams').toWebReadableStream
-    module.exports = braid_fetch
+} else {
+    fetch = window.fetch
+    Headers = window.Headers
 }
```

Comment:
I'm guessing the purpose of lines 15-17 is to make it obvious to the programmer that there are already some variables named `fetch` and `Headers` in the browser Javascript environments?

Reply:
It's actually for the sake of bundlers that this is needed. When a `require` is ambiguously bound (e.g. `fetch` might be either a global variable or a local variable, depending on the if condition) bundlers such as [esbuild](https://github.com/evanw/esbuild) or webpack will create TWO variables, e.g. `fetch` and `fetch2`--one for the local possibility and one for the global possibility. This sounds nice, but unfortunately it results in incorrect code, because the code that uses fetch can only depend on ONE of the variables (either `fetch` or `fetch2`).